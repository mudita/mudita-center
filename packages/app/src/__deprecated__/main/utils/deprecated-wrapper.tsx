/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { FunctionComponent } from "react"

type Element = FunctionComponent<any>

const deprecated = (component: Element, message?: string): Element => {
  if (process.env.NODE_ENV === "development") {
    const basicMessage = `This component is deprecated.`
    if (message) {
      console.warn(`${basicMessage} ${message}`)
    } else {
      console.warn(basicMessage)
    }
  }

  return component
}

export default deprecated
