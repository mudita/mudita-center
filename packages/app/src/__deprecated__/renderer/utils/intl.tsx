/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import React from "react"
import { createIntl } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "App/__deprecated__/renderer/locales/default/en-US.json"
import extractLanguageKeys from "App/__deprecated__/renderer/utils/extract-test-locale"

const testLocale = extractLanguageKeys(localeEn)

export const intl = createIntl({
  locale: translationConfig.defaultLanguage,
  messages: process.env.NODE_ENV === "test" ? testLocale : localeEn,
})

// https://github.com/formatjs/formatjs/issues/1467#issuecomment-543872950
let index = 0
export const textFormatters = {
  b: (str: string) => <strong key={`bold-${++index}`}>{str}</strong>,
  u: (str: string) => <u key={`underline-${++index}`}>{str}</u>,
  i: (str: string) => <em key={`italics-${++index}`}>{str}</em>,
  s: (str: string) => <s key={`strike-${++index}`}>{str}</s>,
}
