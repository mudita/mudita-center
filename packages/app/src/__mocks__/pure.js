/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

jest.mock("pure", () => ({
  ...jest.requireActual("pure"),
  default: {},
}))

const pure = require("pure")

module.exports = pure
