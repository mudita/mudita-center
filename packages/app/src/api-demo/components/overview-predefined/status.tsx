/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { APIFC } from "../../models/api-fc.types"
import React from "react"

interface Field {}

interface DeviceStatusProperties {
  fields?: Field[]
}

export const DeviceStatus: APIFC<DeviceStatusProperties> = ({}) => {
  return (
    <div>
      <h1>Status</h1>
    </div>
  )
}
