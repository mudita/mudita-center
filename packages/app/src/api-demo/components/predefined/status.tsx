import React, { FunctionComponent } from "react"

interface Field {

}

interface DeviceStatusProperties {
  fields?: Field[]
}

export const DeviceStatus: FunctionComponent<DeviceStatusProperties> = ({}) => {
  return <div>
    <h1>Status</h1>
  </div>
}
