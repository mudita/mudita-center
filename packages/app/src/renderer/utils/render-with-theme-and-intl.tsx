/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"
import ReactModal from "react-modal"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
