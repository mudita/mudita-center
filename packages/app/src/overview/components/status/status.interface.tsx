/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { DeviceType } from "@mudita/pure"

export interface StatusProps {
  deviceType: DeviceType | null
  batteryLevel: number
  network?: string
  networkLevel?: number
}
