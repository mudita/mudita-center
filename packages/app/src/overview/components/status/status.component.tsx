/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import React from "react"
import { DeviceType } from "@mudita/pure"
import { defineMessages, FormattedMessage } from "react-intl"
import styled from "styled-components"
import { FunctionComponent } from "Renderer/types/function-component.interface"
import { StatusProps } from "App/overview/components/status/status.interface"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import { letterSpacing, textColor } from "Renderer/styles/theming/theme-getters"
import Card, {
  CardText,
  CardContent,
} from "App/overview/components/card.elements"
import BatteryIcon from "Renderer/components/core/icon/battery-icon.component"
import RangeIcon from "Renderer/components/core/icon/range-icon.component"
import { StatusTestIds } from "App/overview/components/status/status-test-ids.enum"
import { backgroundColor } from "Renderer/styles/theming/theme-getters"

const TextInfo = styled(CardText)`
  p {
    margin-top: 1.2rem;
    letter-spacing: ${letterSpacing("small")}rem;
    color: ${textColor("secondary")};
  }
`
const Stats = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  p {
    margin-top: 0.4rem;
  }
  &:not(:last-child) {
    margin-bottom: 1.6rem;
  }
`

const IconContainer = styled.div`
  background-color: ${backgroundColor("main")};
  margin-right: 1.6rem;
  padding: 0.7rem;
`

const messages = defineMessages({
  tooltipTitle: { id: "module.overview.networkTooltipTitle" },
  tooltipDescription: { id: "module.overview.networkTooltipDescription" },
  battery: { id: "module.overview.phoneBattery" },
  noConnection: { id: "module.overview.phoneNoConnection" },
  network: { id: "module.overview.networkName" },
})

const Status: FunctionComponent<StatusProps> = ({
  deviceType,
  className,
  batteryLevel,
  network,
  networkLevel = 0,
}) => {
  const strength = Math.round(networkLevel * 100)

  return (
    <Card className={className}>
      <TextInfo>
        <Text displayStyle={TextDisplayStyle.SecondaryBoldHeading}>
          {deviceType === DeviceType.MuditaPure ? (
            <FormattedMessage id="module.overview.statusPureTitle" />
          ) : (
            <FormattedMessage id="module.overview.statusHarmonyTitle" />
          )}
        </Text>
      </TextInfo>
      <CardContent>
        <div>
          <Stats>
            <IconContainer>
              <BatteryIcon width={2.6} level={batteryLevel} />
            </IconContainer>
            <div>
              <Text
                displayStyle={TextDisplayStyle.LargeBoldText}
                element={"h2"}
                data-testid={StatusTestIds.BatteryLevel}
              >
                {Math.round(batteryLevel * 100)} %
              </Text>
              <Text
                displayStyle={TextDisplayStyle.SmallFadedText}
                message={messages.battery}
              />
            </div>
          </Stats>
          {deviceType === DeviceType.MuditaPure && (
            <Stats>
              <IconContainer>
                <RangeIcon strength={strength} height={2.6} width={2.6} />
              </IconContainer>
              {network ? (
                <div>
                  <Text
                    displayStyle={TextDisplayStyle.LargeBoldText}
                    data-testid={StatusTestIds.NetworkName}
                  >
                    {network}
                  </Text>
                  <Text
                    displayStyle={TextDisplayStyle.SmallFadedText}
                    message={messages.network}
                  />
                </div>
              ) : (
                <Text
                  displayStyle={TextDisplayStyle.LargeBoldText}
                  message={messages.noConnection}
                />
              )}
            </Stats>
          )}
        </div>
      </CardContent>
    </Card>
  )
}

export default Status
