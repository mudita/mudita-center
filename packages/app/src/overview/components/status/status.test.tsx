/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import "@testing-library/jest-dom/extend-expect"
import React from "react"
import { DeviceType } from "@mudita/pure"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import { StatusProps } from "App/overview/components/status/status.interface"
import Status from "App/overview/components/status/status.component"
import { intl } from "Renderer/utils/intl"

const renderStatus = ({
  deviceType = DeviceType.MuditaPure,
  batteryLevel = 0.75,
  network = "Play",
  networkLevel = 0.75,
}: Partial<StatusProps> = {}) => {
  const outcome = renderWithThemeAndIntl(
    <Status
      deviceType={deviceType}
      batteryLevel={batteryLevel}
      network={network}
      networkLevel={networkLevel}
    />
  )
  return {
    ...outcome,
    getButtons: () => outcome.queryAllByRole("button"),
    getDescription: () =>
      outcome.queryByText(
        intl.formatMessage({ id: "module.overview.networkDescription" })
      ),
  }
}

test("matches snapshot", () => {
  const { container } = renderStatus()
  expect(container).toMatchSnapshot()
})

describe("Device: Mudita pure", () => {
  test("pure info renders properly", () => {
    const { getByText, getByTestId } = renderStatus({
      deviceType: DeviceType.MuditaPure,
    })
    expect(getByText("75 %")).toBeInTheDocument()
    expect(getByText("Play")).toBeInTheDocument()
    expect(
      getByText(intl.formatMessage({ id: "module.overview.phoneBattery" }))
    ).toBeInTheDocument()
    expect(getByTestId("icon-HighRange")).toBeInTheDocument()
  })

  test("renders card title properly", () => {
    const { queryByText } = renderStatus({ deviceType: DeviceType.MuditaPure })
    expect(
      queryByText(intl.formatMessage({ id: "module.overview.statusPureTitle" }))
    ).toBeInTheDocument()
  })
})

describe("Device: Mudita harmony", () => {
  test("harmony info renders properly", () => {
    const { getByText, queryByTestId } = renderStatus({
      deviceType: DeviceType.MuditaHarmony,
    })
    expect(
      getByText(intl.formatMessage({ id: "module.overview.phoneBattery" }))
    ).toBeInTheDocument()
    expect(queryByTestId("icon-HighRange")).not.toBeInTheDocument()
  })

  test("renders card title properly", () => {
    const { queryByText } = renderStatus({
      deviceType: DeviceType.MuditaHarmony,
    })
    expect(
      queryByText(
        intl.formatMessage({ id: "module.overview.statusHarmonyTitle" })
      )
    ).toBeInTheDocument()
  })
})
