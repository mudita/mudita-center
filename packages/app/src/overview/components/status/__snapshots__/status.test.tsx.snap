// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c3 {
  margin: 0;
  font-size: 2.4rem;
  line-height: 0.8;
  font-weight: 700;
  -webkit-letter-spacing: -0.06rem;
  -moz-letter-spacing: -0.06rem;
  -ms-letter-spacing: -0.06rem;
  letter-spacing: -0.06rem;
}

.c8 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.05;
  font-weight: 700;
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
}

.c9 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  color: #6a6a6a;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.6rem;
  width: 2.6rem;
  position: relative;
  font-style: normal;
}

.c7 svg {
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 4rem;
  box-sizing: border-box;
  border-radius: 0.5rem;
  background-color: #ffffff;
}

.c1 {
  grid-area: Text;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-top: auto;
}

.c2 p {
  margin-top: 1.2rem;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  color: #6a6a6a;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 p {
  margin-top: 0.4rem;
}

.c5:not(:last-child) {
  margin-bottom: 1.6rem;
}

.c6 {
  background-color: #fbfbfb;
  margin-right: 1.6rem;
  padding: 0.7rem;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <h2
        class="c3"
      >
        [value] module.overview.statusPureTitle
      </h2>
    </div>
    <div
      class="c4"
    >
      <div>
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <span
              class="c7"
              data-testid="icon-HighBattery"
              height="2.6"
              width="2.6"
            >
              <test-file-stub
                image="test-file-stub"
              />
            </span>
          </div>
          <div>
            <h2
              class="c8"
              data-testid="battery-level"
            >
              75
               %
            </h2>
            <p
              class="c9"
            >
              [value] module.overview.phoneBattery
            </p>
          </div>
        </div>
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <span
              class="c7"
              data-testid="icon-HighRange"
              height="2.6"
              width="2.6"
            >
              <test-file-stub
                image="test-file-stub"
              />
            </span>
          </div>
          <div>
            <p
              class="c8"
              data-testid="network-name"
            >
              Play
            </p>
            <p
              class="c9"
            >
              [value] module.overview.networkName
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
